tools/code-owners-parser/CodeOwnersParser/MatchedCodeOwnerEntry.cs

```diff
@@ -40,145 +64,104 @@ public MatchedCodeOwnerEntry(List<CodeOwnerEntry> entries, string targetPath)
-        /// which then can be checked against using glob matcher.
-        /// </summary>
-        /// <returns>
-        /// Usually 1 glob pattern to include in matching. In one special case
-        /// returns 2 patterns, which happens when the path needs to be interpreted
-        /// both as-is file, or as a directory prefix.
-        /// </returns>
-        private static List<string> ConvertToGlobPatterns(string codeownersPath)
+        private static bool Matches(string targetPath, CodeOwnerEntry entry)
         {
-            codeownersPath = ConvertPrefix(codeownersPath);
-            var patternsToInclude = PatternsToInclude(codeownersPath);
-            return patternsToInclude;
-        }
+            string codeownersPath = entry.PathExpression;
 
-        private static string ConvertPrefix(string codeownersPath)
-        {
-            // Codeowners entry path starting with "/*" is equivalent to it starting with "*".
-            // Note this also covers cases when it starts with "/**".
-            if (codeownersPath.StartsWith("/*"))
-                codeownersPath = codeownersPath.Substring("/".Length);
-
-            // If the codeownersPath doesn't have any slash at the beginning or in the middle,
-            // then it means its start is relative to any directory in the repository,
-            // hence we prepend "**/" to reflect this as a glob pattern.
-            if (!codeownersPath.TrimEnd('/').Contains("/"))
-            {
-                codeownersPath = "**/" + codeownersPath;
-            }
-            // If, on the other hand, codeownersPath has to start at the root, we ensure
-            // it starts with slash to reflect that.
-            else 
-            {
-                if (!codeownersPath.StartsWith("/"))
-                {
-                    codeownersPath = "/" + codeownersPath;
-                }
-                else
-                {
-                    // codeownersPath already starts with "/", so nothing to prepend.
-                }
-            }
+            targetPath = NormalizeTargetPath(targetPath, codeownersPath);
 
-            return codeownersPath;
+            Regex regex = ConvertToRegex(codeownersPath);
+            return regex.IsMatch(targetPath);
         }
 
-        private static List<string> PatternsToInclude(string codeownersPath)
+        private static string NormalizeTargetPath(string targetPath, string codeownersPath)
         {
-            List<string> patternsToInclude = new List<string>();
-
-            if (codeownersPath.EndsWith("/"))
-            {
-                patternsToInclude.Add(ConvertDirectorySuffix(codeownersPath));
-            }
-            else
-            {
-                patternsToInclude.Add(ConvertDirectorySuffix(codeownersPath + "/"));
-                patternsToInclude.Add(codeownersPath);
-            }
-
-            return patternsToInclude;
+            // If the considered CODEOWNERS path ends with "/", it means we can
```

Comment:
I'm not sure I'm following this logic. If we do want to allow cases here would it be better handled with the regex pattern as opposed to altering the target path we are checking?

Reply:
Replaced with [regex manipulation](https://github.com/Azure/azure-sdk-tools/blob/2fd99b4ba2f6ceb30cff343ebb61344931b980d4/tools/code-owners-parser/CodeOwnersParser/MatchedCodeOwnerEntry.cs#L137-L159) instead.